### PR TITLE
Allow disabling team member subscription to SGTM tasks

### PR DESCRIPTION
**Pull request by rpinkerton**
This can help minimize Asana inbox noise when using Github teams as a review or triage marker

Test Plan:

Tested against https://github.com/Asana/codez/pull/195892:

```
rileypinkerton@Riley-Pinkerton-MacBook-Pro SGTM % python
Python 3.9.16 (main, Dec  7 2022, 10:06:04)
[Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import src.github.controller as github_controller; import src.github.graphql.client as graphql_client; import src.github.logic as github_logic
>>> pull_request = graphql_client.get_pull_request("PR_kwDOAr0gsc5QqbFl")
>>> github_logic.pull_request_participants(pull_request)
['binoche9', 'vsiao', 'marikobi', 'fanfan15', 'rpinkerton', 'asopkin', 'kharvd', 'suzyng83209', 'Asana', 'wenchienchen-asana', 'ericrafalovsky']
>>> exit()
rileypinkerton@Riley-Pinkerton-MacBook-Pro SGTM % export SGTM_FEATURE__DISABLE_GITHUB_TEAM_SUBSCRIPTION=true
rileypinkerton@Riley-Pinkerton-MacBook-Pro SGTM % python
Python 3.9.16 (main, Dec  7 2022, 10:06:04)
[Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import src.github.controller as github_controller; import src.github.graphql.client as graphql_client; import src.github.logic as github_logic
>>> pull_request = graphql_client.get_pull_request("PR_kwDOAr0gsc5QqbFl")
>>> github_logic.pull_request_participants(pull_request)
['binoche9', 'suzyng83209', 'ericrafalovsky', 'rpinkerton', 'Asana']
```

Run relevant tests:

```
rileypinkerton@Riley-Pinkerton-MacBook-Pro SGTM % python3 -m unittest test.github.models.test_pull_request
..
----------------------------------------------------------------------
Ran 2 tests in 0.000s

OK
```

Pull Request synchronized with [Asana task](https://app.asana.com/0/0/1204730388775722)

### CODE REVIEW

**Comment by rpinkerton** [comment]
Current dependencies on/for this PR:
* master
  * **PR #146** <a href="https://app.graphite.dev/github/pr/Asana/SGTM/146" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Asana/SGTM/146?utm_source=stack-comment).

File: terraform/variables.tf
Comment (gregeinfrank) [inline terraform/variables.tf:65]: nit:

```suggestion
  description = "'true' if behavior to auto-subscribe github team members is disabled"
```

File: src/github/logic.py
Comment (gregeinfrank) [inline src/github/logic.py:208]: Unfortunately, I think you're going to run into a merge conflict with #143  - since that change is a bit more involved, do you mind re-implementing this change on top of that one? I'm expecting to get it merged tomorrow when @suzyng83209 is back

File: test/github/models/test_pull_request.py
Comment (gregeinfrank) [inline test/github/models/test_pull_request.py:25]: I see you're bypassing the feature flag check by testing one level deeper. I think that's fine - but were you hoping to test this another way and ran into issues with the feature flag? (this is related to your "cargo cult" comment in #144

Comment (rpinkerton) [reply test/github/models/test_pull_request.py:25]: My comment on #144 was mostly about how to test/deploy the the flag from a terraform PoV, my manual testing covers the flag worked as expected from the envvar perspective (but I can add a concrete test for that too if you think it's valuable :) )

Comment (gregeinfrank) [reply test/github/models/test_pull_request.py:25]: Gotcha. Take a look at https://app.asana.com/0/1149418478823393/815237203241118/f (which you'll also want to update once this is merged, if you want to have Asana's SGTM enable this flag).

Let me know if you'd like further help testing this. I think you mainly want to `export SGTM_FEATURE__DISABLE_GITHUB_TEAM_SUBSCRIPTION=true` before opening up your `ipython` shell

**Review by gregeinfrank — APPROVED** [review]
Code LGTM, let's wait on #143  though and then fix merge conflicts after that. I'll re-assign to @suzyng83209 to take a look as well, as the new SGTM AOR

File: src/github/logic.py
Comment (haghdaee) [inline src/github/logic.py:225]: This is interesting. This change leaves CCed reviewers intact. Creating PRs using git pr currently this is what I see:

* if I select a team in the first prompt for reviewer, it picks a random member from the team as the reviewer. like [this](https://github.com/Asana/codez/pull/198536). 
* if I select a team in the second prompt for reviewer, it adds every member of the team, but ALSO adds them as cced like [this](https://github.com/Asana/codez/pull/198536) (which your pr doesn't block notifications for. 

I think the git PR flow would also need to be updated to 
1. take the whole team as the first reviewer prompt
2. not add all reviewers as CCed (or also update this PR to exclude CCed people). 

Another note: I would couple this with [github notifications ](https://app.asana.com/0/1193066022368038/1202528365393009/f)and you're golden!

**Review by haghdaee — CHANGES_REQUESTED** [review]

**Review by haghdaee — APPROVED** [review]
Based on this message from @rpinkerton approving:

```
Ah, you might be a little confused. I made two separate changes:
A change to git pr that enables you to add an actual github team as a reviewer or cc (see https://github.com/Asana/codez/pull/195892) if the team you’re trying to use isn’t in the hardcoded allowlist you’ll get the legacy behavior of just adding all the team members as direct assigns or CCs
SGTM separately denormalized team assigns (which before the previous change never happened anyways) and added everyone as a follower on the task. My change in that repo disables that functionality
```

File: src/github/logic.py
Comment (rpinkerton) [reply src/github/logic.py:225]: Ah, you might be a little confused. I made two separate changes:
* A change to Asana's internal git pr flow that enables you to add an actual github team as a reviewer or cc (see https://github.com/Asana/codez/pull/195892) if the team you’re trying to use isn’t in the hardcoded allowlist you’ll get the legacy behavior of just adding all the team members as direct assigns or CCs
* SGTM separately currently denormalizes team assigns (which before the previous change never happened anyways within Asana) and added everyone as a follower on the task. My change here allows you to disable that functionality

**Comment by rpinkerton** [comment]
@suzyng83209 I'll wait for your stamp here to make sure we feel okay about this functionality :)

**Review by suzyng83209 — APPROVED** [review]

**Pull request merged**
